common/src/verifier.rs

```diff
@@ -33,31 +34,62 @@ pub struct FirmwareImageVerificationEnv<'a, 'b> {
     pub image: &'b [u8],
     pub dma: &'a Dma,
     pub persistent_data: &'a PersistentData,
+    pub image_in_mcu: bool,
+}
+
+impl FirmwareImageVerificationEnv<'_, '_> {
+    fn create_dma_recovery<'a>(soc_ifc: &'a SocIfc, dma: &'a Dma) -> DmaRecovery<'a> {
+        DmaRecovery::new(
+            soc_ifc.recovery_interface_base_addr().into(),
+            soc_ifc.caliptra_base_axi_addr().into(),
+            soc_ifc.mci_base_addr().into(),
+            dma,
+        )
+    }
 }
 
 impl ImageVerificationEnv for &mut FirmwareImageVerificationEnv<'_, '_> {
     /// Calculate 384 digest using SHA2 Engine
     fn sha384_digest(&mut self, offset: u32, len: u32) -> CaliptraResult<ImageDigest384> {
         let err = CaliptraError::IMAGE_VERIFIER_ERR_DIGEST_OUT_OF_BOUNDS;
-        let data = self
-            .image
-            .get(offset as usize..)
-            .ok_or(err)?
-            .get(..len as usize)
-            .ok_or(err)?;
-        Ok(self.sha2_512_384.sha384_digest(data)?.0)
+        if self.image_in_mcu {
+            let dma = FirmwareImageVerificationEnv::create_dma_recovery(self.soc_ifc, self.dma);
+            let result =
+                dma.sha384_mcu_sram(self.sha2_512_384_acc, offset, len, dma::AesDmaMode::None)?;
+            cprintln!(
+                "[rt] SHA384 digest calculated: {}",
+                HexBytes(result.as_bytes())
+            );
```

Comment:
optional: I think we can drop this print to save a little codespace. Mostly because the other digest calculations don't print anything